src/engine/ruleset/decoders/sophos/xg/sophos-antispam.yml

```diff
@@ -0,0 +1,126 @@
+    - event.original: \<<~>><~tmp.payload_message>
+
+normalize:
+  - map:
+      - ~tmp.payload_message: +s_replace/=""/=" "
+    logpar:
+      - ~tmp.payload_message: <~tmp/kv/=/ /"/'>
+  - map:
+    - \@timestamp: +s_concat/$~tmp.date/T/$~tmp.time
+    - destination.bytes: $~tmp.recv_bytes
+    - destination.ip: $~tmp.dst_ip
+    - destination.nat.ip: $~tmp.tran_dst_ip
+    - destination.nat.port: $~tmp.tran_dst_port
+    - destination.packets: $~tmp.recv_pkts
+    - destination.port: $~tmp.dst_port
+    - destination.user.email: $~tmp.to_email_address
+    - event.action: +s_lo/$~tmp.log_subtype
+    - event.category: ["network", "malware", "intrusion_detection"]
+    - event.code: $~tmp.log_id
+    - event.dataset: "sophos.xg"
+    - event.duration: $~tmp.duration
+    - event.end: +s_concat/$~tmp.date/T/$~tmp.time
+    - event.kind: "event"
+    - event.module: "sophos"
+    - event.outcome: "success"
+    - event.reason: $~tmp.reason
+    - event.severity: 6
+    - event.timezone: $~tmp.timezone
+    - event.type: ["allowed", "denied", "connection", "info"]
+    - fileset.name: "xg"
+    - input.type: "log"
+    - log.level: $~tmp.priority
+    - network.bytes: $~tmp.sent_bytes
+    - network.protocol: $~tmp.log_component
+    - network.transport: $~tmp.protocol
+    - observer.product: "XG"
+    - observer.serial_number: $~tmp.device_id
+    - observer.type: "firewall"
+    - observer.vendor: "Sophos"
+    - related.hosts: [$~tmp.src_domainname]
+    - related.ip: [$~tmp.src_ip, $~tmp.dst_ip]
+    - service.type: "Sophos"
+    - sophos.xg.app_is_cloud: $~tmp.app_is_cloud
+    - sophos.xg.appfilter_policy_id: $~tmp.appfilter_policy_id
+    - sophos.xg.application_risk: $~tmp.application_risk
+    - sophos.xg.appresolvedby: $~tmp.appresolvedby
+    - sophos.xg.av_policy_name: $~tmp.av_policy_name
+    - sophos.xg.connevent: $~tmp.connevent
+    - sophos.xg.connid: $~tmp.connid
+    - sophos.xg.device: $~tmp.device
+    - sophos.xg.device_name: $~tmp.device_name
+    - sophos.xg.dst_country_code: $~tmp.dst_country_code
+    - sophos.xg.dst_zone_type: $~tmp.dstzonetype
+    - sophos.xg.email_subject: $~tmp.email_subject
+    - sophos.xg.ether_type: $~tmp.ether_type
+    - sophos.xg.fw_rule_id: $~tmp.fw_rule_id
+    - sophos.xg.hb_health: $~tmp.hb_health
+    - sophos.xg.iap: $~tmp.iap
+    - sophos.xg.ips_policy_id: $~tmp.ips_policy_id
+    - sophos.xg.log_component: $~tmp.log_component
+    - sophos.xg.log_id: $~tmp.log_id
+    - sophos.xg.log_subtype: $~tmp.log_subtype
+    - sophos.xg.log_type: $~tmp.log_type
+    - sophos.xg.mailid: $~tmp.mailid
+    - sophos.xg.mailsize: $~tmp.mailsize
+    - sophos.xg.priority: $~tmp.priority
+    - sophos.xg.quarantine_reason: $~tmp.quarantine_reason
+    - sophos.xg.spamaction: $~tmp.spamaction
+    - sophos.xg.src_country_code: $~tmp.src_country_code
+    - sophos.xg.src_zone_type: $~tmp.srczonetype
+    - sophos.xg.status: $~tmp.status
+    - sophos.xg.user_name: $~tmp.user_name
```

Comment:
Have you considered merging the fields instead of copying the values?

Reply:
I actually tried to do the direct mapping in the parser's logpar, but there were fields that didn't always come in the same order. Maybe I would have done that only until the part of the event that the fields were sorted and then just use the kv_parser in the rest of the event.